src/components/progressCircular/progress-circular.js

```diff
@@ -87,43 +89,76 @@ function MdProgressCircularDirective($mdConstant, $mdTheming, $mdUtil) {
   function postLink(scope, element, attr) {
     $mdTheming(element);
 
-    var circle = element[0];
+    var circle = element;
     var spinnerWrapper =  angular.element(element.children()[0]);
-
     var lastMode, toVendorCSS = $mdUtil.dom.animator.toCss;
 
-    // Update size/scaling of the progress indicator
-    // Watch the "value" and "md-mode" attributes
-
-    circle.style[$mdConstant.CSS.TRANSFORM] = 'scale(' + getDiameterRatio() + ')';
-
-    attr.$observe('value', function(value) {
-      var percentValue = clamp(value);
-      element.attr('aria-valuenow', percentValue);
+    updateScale();
+    validateMode();
+    watchAttributes();
 
-      if (attr.mdMode == "determinate") {
-        animateIndicator(percentValue);
-      }
-    });
+    /**
+     * Watch the value and md-mode attributes
+     */
+    function watchAttributes() {
+     attr.$observe('value', function(value) {
+           var percentValue = clamp(value);
+           element.attr('aria-valuenow', percentValue);
+
+           if (mode() == MODE_DETERMINATE) {
+             animateIndicator(percentValue);
+           }
+         });
+     attr.$observe('mdMode',function(mode){
+       switch( mode ) {
+         case MODE_DETERMINATE:
+         case MODE_INDETERMINATE:
+           spinnerWrapper.removeClass('ng-hide');
+
+           // Inject class selector instead of attribute selector
+           // (@see layout.js changes for IE performance issues)
+
+           if ( lastMode ) spinnerWrapper.removeClass( lastMode );
+                lastMode = "md-mode-" + mode;
+           if ( lastMode ) spinnerWrapper.addClass( lastMode );
```

Comment:
We need to cache the lastMode for use in the `default` case:

```js
switch( mode ) {
          case MODE_QUERY:
          case MODE_BUFFER:
          case MODE_DETERMINATE:
          case MODE_INDETERMINATE:
            container.removeClass('ng-hide');

            // Inject class selector instead of attribute selector
            // (@see layout.js changes for IE performance issues)

            container.removeClass( lastMode );
            lastMode = "md-mode-" + mode;
            container.addClass( lastMode );

            break;
          default:
            container.addClass('ng-hide');
            container.removeClass( lastMode );
            lastMode = undefined;
            break;
        }
```